cmd/lakefs/cmd/gc.go

```diff
@@ -0,0 +1,310 @@
+	return ref.NewOrderedCommitIterator(ctx, r.store, repository, true)
+}
+
+func (r *RefManager) ListCommits(ctx context.Context, repository *graveler.RepositoryRecord) (graveler.CommitIterator, error) {
+	return ref.NewOrderedCommitIterator(ctx, r.store, repository, false)
+}
+
+func (r *RefManager) GetCommit(ctx context.Context, repository *graveler.RepositoryRecord, commitID graveler.CommitID) (*graveler.Commit, error) {
+	var data graveler.CommitData
+
+	if _, err := kv.GetMsg(ctx, r.store, r.repositoryID, []byte(graveler.CommitPath(commitID)), &data); err != nil {
+		return nil, fmt.Errorf("%s: %w", commitID, err)
+	}
+	commit := graveler.CommitFromProto(&data)
+	return commit, nil
+}
+
+var simulateCmd = &cobra.Command{
+	Use:   "simulate rules.json partition-dump.json",
+	Short: "Simulate a GC run on locally-available files",
+	Long: `Run garbage collection on locally-available files.
+
+First arg "rules.json" is a file containing the JSON retention rules.  Second arg
+"partition-dump" is a file containing the entire JSON scan of the branch partition on KV.
+Create it with "lakefs kv scan <reponame>-ID".  To discover the ID try "lakefs kv scan graveler"
+and look for the instance_uid field.
+`,
+	//nolint:mnd
+	Args: cobra.ExactArgs(2), Run: func(cmd *cobra.Command, args []string) {
+		ctx := cmd.Context()
+
+		output := os.Stdout
+		outputFilename, err := cmd.Flags().GetString("output")
+		if err != nil {
+			printMsgAndExit(fmt.Errorf("output: %w", err))
+		}
+		if outputFilename != "" {
+			output, err = os.Create(outputFilename)
+			if err != nil {
+				printMsgAndExit(fmt.Errorf("open %s: %w", outputFilename, err))
+			}
+			defer output.Close()
+		}
+
+		cpuProfileFilename, err := cmd.Flags().GetString("cpuprofile")
+		if err != nil {
+			printMsgAndExit(fmt.Errorf("cpuprofile: %w", err))
+		}
+		var cpuProfileFile *os.File
+		if cpuProfileFilename != "" {
+			cpuProfileFile, err = os.Create(cpuProfileFilename)
+			if err != nil {
+				printMsgAndExit(fmt.Errorf("open %s: %w", cpuProfileFilename, err))
+			}
+			defer cpuProfileFile.Close()
+			// Will start profiling later.
+		}
+		memProfileFilename, err := cmd.Flags().GetString("memprofile")
+		if err != nil {
+			printMsgAndExit(fmt.Errorf("memprofile: %w", err))
+		}
+		var memProfileFile *os.File
+		if memProfileFilename != "" {
+			memProfileFile, err = os.Create(memProfileFilename)
+			if err != nil {
+				printMsgAndExit(fmt.Errorf("open %s: %w", memProfileFilename, err))
+			}
+			defer memProfileFile.Close()
+		}
+
+		// Allow profiling over a web-server, too.
+		cfg := LoadConfig()
+		baseCfg := cfg.GetBaseConfig()
+		pprofHandler := httputil.ServePPROF("/_pprof/")
+		server := http.Server{
+			Addr:              baseCfg.ListenAddress,
+			Handler:           pprofHandler,
+			ReadHeaderTimeout: pprofReadHeaderTimeout,
+		}
+		go func() {
+			_ = server.ListenAndServe() // this server never exits.
+		}()
```

Comment:
its a cli, any real use of this?
we should use profile.Start/Stop (optional flag) to enable profile on the code we like to collect profile information.